src/codeu/chat/ClientMain.java

```diff
@@ -44,8 +44,9 @@ public static void main(String [] args) {
     final RemoteAddress address = RemoteAddress.parse(args[0]);
 
     final ConnectionSource source = new ClientConnectionSource(address.host, address.port);
-    final Controller controller = new Controller(source);
+
     final View view = new View(source);
+    final Controller controller = new Controller(source, view.getServerPublicKey());
```

Comment:
Could you clarify which part you're referring to? Creating the Controller after the View?